### DISCUSSION TITLE
How to subscribe Render(DrawingContext context) from a control?

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#7909] [Q&A] [answered]
I have a `ContentPresenter` inside my `Window`

From `Window` using code, how can i subscribe the ContentPresenter `Render(DrawingContext context)` and draw inside it?
In old days and WinFroms it was the `Paint` event, but i can't find a similar way on Avalonia.

**Comment by maxkatz6** [comment, marked as answer]
You override Render method.
```csharp
public class MyControl : Control
{
    public override void Render( *** ) { *** }
}
```

**Reply by sn4k3** [reply]
I mean, do that but from outside the control scope, nor have to derived the control to do it. 
In other words, subscribe an event and handle from there, for example: https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control.paint?view=windowsdesktop-6.0.

**Reply by maxkatz6** [reply]
No, it's not possible
